### ISSUE TITLE
Resynchronize with pthreads-win32

### ISSUE THREAD

**Issue by dmik** [bitwiseworks/pthread-os2#10]
Our implementation is based on [pthreads4w](https://sourceforge.net/projects/pthreads4w/), formerly known as [pthreads-win32](https://sourceware.org/pthreads-win32/). There has been a lot of changes since we last synced in 2010 (see e4528d98142223de3dde58f79e175519eade6bf9, 171b54d8c2ae63068b892acddf16c80469c5949c). In particular, the latest win32 implementation supports PTHREAD_MUTEX_ROBUST and other nice things.

Once this is done, #9 should be not necessary any more (and could be undone).

Note urgent (we don't have projects strongly dependent on new stuff) so postponed.
